### DISCUSSION TITLE
wire:click event child emit to parent error - Uncaught (in promise) undefined

### DISCUSSION THREAD

**Discussion by elizdlt202** [livewire/livewire#6570] [7. Bugs (archived)]
Please help....

laravel v9 
livewire v3

I want the parent component to listen to the dispatched event from the child.

Upon the first click... it works as expected and I have a counter..
but once I click again.... I get a console.log error and the counter does not work.

what am I doing wrong here:

child:
```
<div class="flex flex-row place-content-end items-center">
    
    <button
        wire:click="$parent.incrementStep()"
        id='next-step'
        class="inline-flex">
        Next Step
        
    </button>

</div>
```

parent
```
 public $count = 1;



    #[On('incrementStep')]
    public function incrementStep()
    {
        Log::info('you made it!!!!');
        $this->count++;
    }
```

**Comment by simonmilz** [comment]
I have the same error. First request works fine, then I get an error in my console: "Uncaught (in promise) undefined"

**Comment by farizhuzairi** [comment]
Since previous versions this has happened (implementation of "emit"), even up to the latest version when it's implementation of "dispatch". I don't know what's wrong?

event components:
`    public function selectHub(): void
    {
        $this->dispatch('changeMenu', visit: 'hubs')->to(Hubs::class);
    }
`

listener component:

`    #[On('changeMenu')]
    public function selectMenu(string $visit = '')
    {
        $this->hubContainer = false;
        $this->connectivityContainer = false;

        switch ($visit){
            case 'hubs':
                $this->hubContainer = true;
                $blade = 'hubs';
                break;
            case 'connectivity':
                $this->connectivityContainer = true;
                $blade = 'connectivity';
                break;
            default:
                $this->hubContainer = true;
                $blade = 'hubs';
                break;
        }

        $this->pageView = self::BLADEVIEW . $blade;
    }`
